metaflow/datastore/task_datastore.py

```diff
@@ -353,15 +352,13 @@ def load_artifacts(self, names):
                     "Python 3.4 or later is required to load artifact '%s'" % name
                 )
             else:
-                sha = self._objects[name]
-                sha_to_names[sha] = name
-                to_load.append(sha)
+                to_load.append(self._objects[name])
         # At this point, we load what we don't have from the CAS
         # We assume that if we have one "old" style artifact, all of them are
         # like that which is an easy assumption to make since artifacts are all
         # stored by the same implementation of the datastore for a given task.
-        for sha, blob in self._ca_store.load_blobs(to_load):
-            yield sha_to_names[sha], pickle.loads(blob)
+        for name, (_, blob) in zip(names, self._ca_store.load_blobs(to_load)):
```

Comment:
this logic assumes that `load_blobs` returns the blobs in the order of `to_load`, which could break in the future and lead to very subtle bugs. You could grab the `(key, blob)` pair in the return value and use it to produce names